bagua/torch_api/distributed.py

```diff
@@ -147,6 +147,7 @@ def with_bagua(  # pytype: disable=module-attr
         self,
         optimizers: List[torch.optim.Optimizer],
         algorithm: "bagua.torch_api.algorithms.Algorithm",
+        communication_interval: int = 1,
```

Comment:
this should be in the algorithm, since it does not make sense for some algorithms like gradient allreduce but tends to get misused.

Reply:
oh. it sounds better.